src/atomate2/common/flows/mpmorph.py

```diff
@@ -0,0 +1,493 @@
+"""Define code agnostic MPMorph flows.
+
+This file generalizes the MPMorph workflows of
+https://github.com/materialsproject/mpmorph
+originally written in atomate for VASP only to a more general
+code agnostic form.
+
+For information about the current flows, contact:
+- Bryant Li (@BryantLi-BLI)
+- Aaron Kaplan (@esoteric-ephemera)
+- Max Gallant (@mcgalcode)
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, Literal
+
+import numpy as np
+from jobflow import Flow, Maker, Response, job
+
+from atomate2.common.jobs.eos import MPMorphPVPostProcess, _apply_strain_to_structure
+
+if TYPE_CHECKING:
+    from pathlib import Path
+    from typing import Any
+
+    from jobflow import Job
+    from pymatgen.core import Structure
+    from typing_extensions import Self
+
+    from atomate2.common.jobs.eos import EOSPostProcessor
+
+
+@dataclass
+class EquilibriumVolumeMaker(Maker):
+    """
+    Equilibrate structure using NVT + EOS fitting.
+
+    Parameters
+    ----------
+    name : str = "Equilibrium Volume Maker"
+        Name of the flow
+    md_maker : Maker
+        Maker to perform NVT MD runs
+    postprocessor : atomate2.common.jobs.eos.EOSPostProcessor
+        Postprocessing step to fit the EOS
+    initial_strain : float | tuple[float,float] = 0.2
+        Initial percentage linear strain to the apply to the structure
+    min_strain : float, default = 0.5
+        Minimum absolute percentage linear strain to apply to the structure
+    max_attempts : int | None = 20
+        Number of times to continue attempting to equilibrate the structure.
+        If None, the workflow will not terminate if an equilibrated structure
+        cannot be determined.
+    """
+
+    name: str = "Equilibrium Volume Maker"
+    md_maker: Maker | None = None
```

Comment:
`md_maker` is allowed to be `None` but looks like `make` will fail if it is? why not make this a required arg?

Reply:
See above, this has a `__post_init__` for checking this is not `None` now

Reply:
i think adding a `__post_init__` can be considered boilerplate. if you make `md_maker` a required arg and the user doesn't pass it, you `dataclass` should raise a similar error message 

```
    def __post_init__(self) -> None:
        if self.md_maker is None:
            raise ValueError("You must specify `md_maker` to use this flow.")
```

Reply:
Right - that's there already

Reply:
i know, i'm suggesting removing it and changing

```diff
- md_maker: Maker | None = None
+ md_maker: Maker
```
to make `md_maker` a required arg

Reply:
Ahhh yes sorry! Changed as requested, leaving in the `__post_init__`'s as well tho